src/v/cluster/topics_frontend.cc

```diff
@@ -421,11 +427,12 @@ ss::future<topic_result> topics_frontend::do_create_topic(
         co_return topic_result(assignable_config.cfg.tp_ns, validation_err);
     }
 
-    // TODO: implement read replicas, see
-    // https://github.com/redpanda-data/redpanda/issues/4736
     if (assignable_config.is_read_replica()) {
-        return ss::make_ready_future<topic_result>(topic_result(
-          assignable_config.cfg.tp_ns, errc::topic_invalid_config));
+        _read_replica_manager = ss::make_shared<cluster::read_replica_manager>(
```

Comment:
nit: here you can probably just create `read_replica_manager` on a stack, use it to call `propagate_remote_properties` once and let it be deleted. As for now I don't see why do you need it as a field of the object. Maybe some other changes are planned that will use the `_read_replica_manager` field?

Reply:
creating on stack sounds good